lib/rspec/support/source/token.rb

```diff
@@ -0,0 +1,149 @@
+RSpec::Support.require_rspec_support 'source/location'
+
+module RSpec
+  module Support
+    class Source
+      # @private
+      # A wrapper for Ripper token which is generated with `Ripper.lex`.
+      class Token
+        CLOSING_TYPES_BY_OPENING_TYPE = {
+          :on_lbracket    => :on_rbracket,
+          :on_lparen      => :on_rparen,
+          :on_lbrace      => :on_rbrace,
+          :on_heredoc_beg => :on_heredoc_end
+        }.freeze
+
+        CLOSING_KEYWORDS_BY_OPENING_KEYWORD = {
+          'def' => 'end',
+          'do'  => 'end',
+        }.freeze
+
+        # This is a duplicate of Ripper::SCANNER_EVENTS.
+        # We don't use Ripper::SCANNER_EVENTS here because we don't want load Ripper
+        # until it's really needed to parse sources.
+        RIPPER_SCANNER_EVENTS = [
+          :CHAR,
+          :__end__,
+          :backref,
+          :backtick,
+          :comma,
+          :comment,
+          :const,
+          :cvar,
+          :embdoc,
+          :embdoc_beg,
+          :embdoc_end,
+          :embexpr_beg,
+          :embexpr_end,
+          :embvar,
+          :float,
+          :gvar,
+          :heredoc_beg,
+          :heredoc_end,
+          :ident,
+          :ignored_nl,
+          :imaginary,
+          :int,
+          :ivar,
+          :kw,
+          :label,
+          :label_end,
+          :lbrace,
+          :lbracket,
+          :lparen,
+          :nl,
+          :op,
+          :period,
+          :qsymbols_beg,
+          :qwords_beg,
+          :rational,
+          :rbrace,
+          :rbracket,
+          :regexp_beg,
+          :regexp_end,
+          :rparen,
+          :semicolon,
+          :sp,
+          :symbeg,
+          :symbols_beg,
+          :tlambda,
+          :tlambeg,
+          :tstring_beg,
+          :tstring_content,
+          :tstring_end,
+          :words_beg,
+          :words_sep
+        ].freeze
```

Comment:
Here too I'd like to see us get away from a constant we have to keep in sync with ripper...and I'm also curious as to why we need it here but did not in rspec-core.